### PR TITLE
Add split route support in all adapters

### PR DESCRIPTION
**Pull request by thal0x**
Updates adapter contracts to support multiple routes. Excluding:
- Astroport added in #99 
- Lido Satellite, doesn't seem relevant

Updates `osmosis-std` to access split route types

**Note:** I checked the Dexter router code and they properly track balances between swaps so the edge case found in #99 won't be an issue.

### CODE REVIEW

**Comment by NotJeremyLiu** [comment]
> Lido Satellite, doesn't seem relevant

Don't we have to change interfaces for all swap adapters no matter what? How would this be called by the entrypoint without a change in interface?

File: contracts/adapters/swap/dexter/src/contract.rs [hunk 1]
```diff
@@ -181,44 +155,51 @@ fn execute_swap(
...
-    };
+        let dexter_router_msg = RouterExecuteMsg::ExecuteMultihopSwap {
+            requests: hop_swap_requests,
+            recipient: None,
+            offer_amount: route.offer_asset.amount(),
+            // doing this since we would validate it anyway in the entrypoint contract from where swap adapter is called
+            minimum_receive: None,
+        };
+
+        // let denom_in = route.operations.first().unwrap().denom_in.clone();
```
Comment (NotJeremyLiu) [inline contracts/adapters/swap/dexter/src/contract.rs:185]: remove comment

**Comment by thal0x** [comment]
> > Lido Satellite, doesn't seem relevant
> 
> @thal0x Don't we have to change interfaces for all swap adapters no matter what? How would this be called by the entrypoint without a change in interface?

The interface was already updated this PR just changes the logic to not error on multiple routes

File: contracts/adapters/swap/dexter/src/contract.rs [hunk 2]
```diff
@@ -128,25 +119,9 @@ pub fn execute(
...
-            // validate that there's at least one swap operation
-            if operations.is_empty() {
-                return Err(ContractError::SwapOperationsEmpty);
-            }
-
-            let coin = one_coin(&info)?;
+            one_coin(&info)?;
 
-            // validate that the one coin is the same as the first swap operation's denom in
-            if coin.denom != operations.first().unwrap().denom_in {
```
Comment (NotJeremyLiu) [inline contracts/adapters/swap/dexter/src/contract.rs:145]: Do we do validation elsewhere that makes the validations you deleted no longer be relevant? I don't seem to see any validation but may be missing it

Comment (NotJeremyLiu) [reply contracts/adapters/swap/dexter/src/contract.rs:145]: Oh we do validation in entrypoint so we shouldn't need this... I see

File: contracts/adapters/swap/white-whale/src/contract.rs [hunk 3]
```diff
@@ -159,22 +145,40 @@ fn execute_swap(
         return Err(ContractError::Unauthorized);
     }
 
+    // reset the pre swap out asset amount
+    PRE_SWAP_OUT_ASSET_AMOUNT.save(deps.storage, &Uint128::new(0))?;
```
Comment (NotJeremyLiu) [inline contracts/adapters/swap/white-whale/src/contract.rs:149]: Same comment I had on the astroport version

**Review by NotJeremyLiu — APPROVED** [review]
left some comments

**Pull request merged**